### ISSUE TITLE
[JavaScript] Regression in 2.3 using `multi` parameters (Cannot read property 'buildCollectionParam' of undefined)

### ISSUE THREAD

**Issue by frol** [swagger-api/swagger-codegen#5358]
##### Description

@michaelkourlas @wing328 After I upgraded from 2.2 to 2.3, all calls with `multi` parameters fail. DevTools says that `this.apiClient` is undefined [here](https://github.com/swagger-api/swagger-codegen/blob/c35fdc3bdaf9cef413104f9c7d34d76ec9fd0299/modules/swagger-codegen/src/main/resources/Javascript/ApiClient.mustache#L394):

```
TypeError: Cannot read property 'buildCollectionParam' of undefined
```

Here is the context:

```
collectionQueryParams: Object
  columns: Object
    collectionFormat: "multi"
    value: undefined
  filter_by: Object
contentType: undefined
contentTypes: Array(1)
  0: "application/json"
  length: 1
formParams: Object
headerParams: Object
httpMethod: "GET"
key: "columns"
param: Object
  collectionFormat: "multi"
  value: undefined
path: "/data/transformations/{data_transformation_id}/data.{export_format}"
pathParams: Object
queryParams: Object
request: Request
returnType: function ()
this: exports
url: "http://localhost:5000/api/v1/data/transformations/6/data.json"
value: undefined
```

##### Swagger-codegen version

It is a regression in 2.3.

##### Swagger declaration file content or url

This is the relevant part of my Swagger conf:

```json
"parameters": [
                    {
                        "items": {
                            "type": "string"
                        },
                        "type": "array",
                        "description": "Which columns to export",
                        "required": false,
                        "collectionFormat": "multi",
                        "name": "columns",
                        "in": "query"
                    },
                    {
                        "items": {
                            "type": "string"
                        },
                        "type": "array",
                        "description": "a list of filtering rules",
                        "required": false,
                        "collectionFormat": "multi",
                        "name": "filter_by",
                        "in": "query"
                    }
                ],
```

##### Steps to reproduce

1. Generate a client with parameters of `multi` `collectionFormat`
2. Try to make a call

##### Related issues

The change was introduced in https://github.com/swagger-api/swagger-codegen/pull/4725

##### Suggest a Fix

I fail to understand what should [the `else` branch](https://github.com/swagger-api/swagger-codegen/blob/c35fdc3bdaf9cef413104f9c7d34d76ec9fd0299/modules/swagger-codegen/src/main/resources/Javascript/ApiClient.mustache#L394) do, so I just removed it completely and it works for me, but I don't think it is a proper fix.

**Comment by michaelkourlas** [comment]
@frol That else branch is used to parse collection query parameters with a type other than `csv`. Removing it is indeed not a proper fix.

It looks like I made a typo in that line. It should read
`queryParams[key] = this.buildCollectionParam(param.value, param.collectionFormat);`
rather than
`queryParams[key] = this.apiClient.buildCollectionParam(param.value, param.collectionFormat);`

I'll make a PR with this fix.

**Comment by michaelkourlas** [comment]
I am surprised, though, that the tests didn't catch this. I guess we don't test requests with different collection query parameter types.

**Comment by wing328** [comment]
Closed via #5362
